Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -230,6 +230,13 @@ CookieStore::~CookieStore()
     m_mainThreadBridge->detach();
 }
 
+static URL creationURL(ScriptExecutionContext& context)
+{
+    if (RefPtr document = dynamicDowncast<Document>(context))
+        return document->cookieURL();
+    return context.creationURL();
```

Comment:
We should probably just rename `ScriptExecutionContext:: creationURL()` to `cookieURL()`, then we don't need to check here.
The [creation URL](https://html.spec.whatwg.org/multipage/webappapis.html#concept-environment-creation-url) is represented by `ScriptExecutionContext::url()` for non-document context.

Reply:
That's a good suggestion--we added `ScriptExecutionContext:: creationURL()` specifically so that ServiceWorkerGlobalScope would have a `creationURL()` function just like Document to be consistent for getting the url needed for cookies (and it just used `ScriptExecutionContext::url()` under the hood). Now that Document uses `cookieURL()`, it's better to do the same for ServiceWorkerGlobalScope too. I've renamed it `ScriptExecutionContext::cookieURL()`.